src/Microsoft.PowerShell.Commands.Utility/commands/utility/Measure-Object.cs

```diff
@@ -820,9 +820,12 @@ protected override void EndProcessing()
                 Statistics stat = _statistics[propertyName];
                 if (stat.count == 0 && Property != null)
                 {
-                    // Why are there two different ids for this error?
-                    string errorId = (IsMeasuringGeneric) ? "GenericMeasurePropertyNotFound" : "TextMeasurePropertyNotFound";
-                    WritePropertyNotFoundError(propertyName, errorId);
+                    if (Context.EngineSessionState.CurrentScope.StrictModeVersion?.Major != 0) 
```

Comment:
I think you should call `Context.IsStrictVersion(2)`, see the [source code here](https://github.com/PowerShell/PowerShell/blob/87b61a04522b45c01513b7424a14d1362235c46c/src/System.Management.Automation/engine/ExecutionContext.cs#L130-L153).

The reason to pass in `2` as the `majorVersion` for that call is that PowerShell reports error on accessing non-existing property only starting from strict mode version 2:

![image](https://user-images.githubusercontent.com/127450/149246240-6387c160-4926-4538-b64b-e3649d6c4838.png)

Reply:
I have just committed a change addressing this.